source/supported-hardware/mtl-mcl.rst

```diff
@@ -0,0 +1,139 @@
+* DediProg SF600 programmer
+* Linux host or windows host (see :ref:`running-on-linux` or :ref:`running-on-windows` for details)
+* Internet access
+
+.. |Intel McLaren Island Reference Design Board| raw:: html
+
+   <a href="https://www.seavo.com/en/products/products-info_itemid_558.html" target="_blank">Intel McLaren Island Reference Design Board</a>
+
+.. |instructions| raw:: html
+
+   <a href="https://wiki.up-community.org/BIOS_chip_flashing_on_UP_Squared" target="_blank">instructions</a>
+
+Board Setup
+^^^^^^^^^^^^^^^^^
+
+.. image:: /images/mcl-setup.jpg
+   :width: 600
+   :alt: |MTL-MCL| Board Setup
+   :align: center
+
+
+Before You Start
+^^^^^^^^^^^^^^^^^
+
+.. warning:: As you plan to reprogram the SPI flash, it's a good idea to backup the pre-installed BIOS image first.
+
+
+Boot the board and enter BIOS setup menu to get familiar with the board features and settings.
+
+Debug UART
+^^^^^^^^^^^
+
+For |MTL| platforms, serial port connector location can be found from the above table for each supported target board.
+
+.. note:: Configure host PuTTY or minicom to 115200bps, 8N1, no hardware flow control.
+
+Building
+^^^^^^^^^^
+
+|Intel McLaren Island Reference Design Board| is based on Intel |MTL|. To build::
+
+    python BuildLoader.py build mtl
+
+The output images are generated under ``Outputs`` directory.
+
+Stitching
+^^^^^^^^^^
+
+Stitch |SPN| images with factory BIOS image using the stitch tool::
+
+    python Platform/MeteorlakeBoardPkg/Script/StitchLoader.py -i <BIOS_IMAGE_NAME> -s Outputs/mtl/SlimBootloader.bin -o <SBL_IFWI_IMAGE_NAME> -p 0xAA00001F
+
+    <BIOS_IMAGE>     : Input file. Factory BIOS extracted from McLaren Island board.
+    <SBL_IFWI_IMAGE> : Output file. New IFWI image with SBL in BIOS region.
+    -p <value>       : 4-byte platform data for platform ID (e.g. 1F) and debug UART port index (e.g. 00).
+
+.. Note:: StitchLoader.py script works only if Boot Guard in the base image is not enabled, and the silicon is not fused with Boot Guard enabled.
+          If Boot Guard is enabled, please use StitchIfwi.py script instead.
+
+See :ref:`stitch-tool` on how to stitch the IFWI image with |SPN|.
+
+
+Slimbootloader binary for capsule
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+A capsule image could ecapsulate a Slimbootloader image to be used in firmware update mechanism. More information is described in :ref:`firmware-update`.
+
+For the process running properly, an IFWI image programmed on board with TOP SWAP size configuration is required.
+
+.. note:: Enabling TOP SWAP size configuration requires additional firmware components and tools.
+
+Creating Slimbootloader binary for capsule image requires the following steps:
+
+Build |SPN| for |UPX12RPLP|::
+
+  python BuildLoader.py build rplp
+
+Edit the 4-byte platform data in Slimbootloader image by *Hexedit* or equivalent hexdecimal editor.
+
+Go to top of TOP SWAP A at address 0xCFFFF4, edit ``14 01 00 AA`` as below
```

Comment:
@randylintw should edit 1f 00 00 AA, fyi

Reply:
remove it :)